linkml/generators/pythongen.py

```diff
@@ -126,14 +127,135 @@ def visit_type(self, typ: TypeDefinition) -> None:
-        enumimports = (
-            ""
-            if self.genmeta
-            else "from linkml_runtime.linkml_model.meta import EnumDefinition, PermissibleValue, PvFormulaOptions\n"
+        if not self.genmeta:
+            all_imports = all_imports + Import(
+                module="linkml_runtime.linkml_model.meta",
+                objects=[
+                    ObjectImport(name="EnumDefinition"),
+                    ObjectImport(name="PermissibleValue"),
+                    ObjectImport(name="PvFormulaOptions"),
+                ],
+            )
+        # linkml imports
+        all_imports = (
+            all_imports
+            + Import(
+                module="linkml_runtime.utils.slot",
+                objects=[
+                    ObjectImport(name="Slot"),
+                ],
+            )
+            + Import(
+                module="linkml_runtime.utils.metamodelcore",
+                objects=[
+                    ObjectImport(name="empty_list"),
+                    ObjectImport(name="empty_dict"),
+                    ObjectImport(name="bnode"),
+                ],
+            )
+            + Import(
+                module="linkml_runtime.utils.yamlutils",
+                objects=[
+                    ObjectImport(name="YAMLRoot"),
+                    ObjectImport(name="extended_str"),
+                    ObjectImport(name="extended_float"),
+                    ObjectImport(name="extended_int"),
+                ],
+            )
+            + Import(
+                module="linkml_runtime.utils.dataclass_extensions_376",
+                objects=[
+                    ObjectImport(name="dataclasses_init_fn_with_kwargs"),
+                ],
+            )
+            + Import(
+                module="linkml_runtime.utils.formatutils",
+                objects=[
+                    ObjectImport(name="camelcase"),
+                    ObjectImport(name="underscore"),
+                    ObjectImport(name="sfx"),
+                ],
+            )
+        )
+
+        # handler import
+        all_imports = all_imports + Import(
+            module="linkml_runtime.utils.enumerations", objects=[ObjectImport(name="EnumDefinitionImpl")]
+        )
+        # other imports
+        all_imports = (
+            all_imports
+            + Import(
+                module="rdflib",
+                objects=[
+                    ObjectImport(name="Namespace"),
+                    ObjectImport(name="URIRef"),
+                ],
+            )
+            + Import(
+                module="linkml_runtime.utils.curienamespace",
+                objects=[
+                    ObjectImport(name="CurieNamespace"),
+                ],
+            )
         )
-        handlerimport = "from linkml_runtime.utils.enumerations import EnumDefinitionImpl"
+
+        if self._import_field():
+            all_imports = all_imports + Import(module="dataclasses", objects=[ObjectImport(name="field")])
```

Comment:
might as well just always import this, since importing one object from a module means executing the whole module, this wouldn't be a perf hit. pydanticgen uses a system of build results to handle imports that are needed in specific cases and not others, but retrofitting that into the pythongen would be a lot of work.

Reply:
I agree that the complexity of `_import_field` might not be worth. You will only get linter warnings on the generated code, but you get anyway so many other warnings 😄 I will change it as you propose.

Reply:
I like @Silvanoc's approach on generating only the necessary imports but each point of view has its pros and cons 😇

Reply:
i mean yeah i designed a whole system of build result models with the primary purpose of only including necessary imports for pydanticgen, so of course i prefer doing that, just saying that gating with private methods just doesn't scale that well

Reply:
Although I appreciate @vincentkelleher support on my approach. I have to agree with @sneakers-the-rat that saving that `field` import is not worth the added complexity. So I will go for the always import `field` approach.

Reply:
don't mean to be a downer, also am indifferent on this change, defer to the author :)

Reply:
I have revised my decision. Although I am not happy with the complexity added with `_import_field`, importing always `dataclasses.field` results in a cascade of snapshots needing updating. I have decided to leave `_import_field` there... At least for the time being. I will consider afterwards removing it.